packages/blockchain/src/blockchain.ts

```diff
@@ -1173,17 +1177,17 @@ export class Blockchain implements BlockchainInterface {
     switch (this._common.consensusAlgorithm()) {
```

Comment:
@acolytec3 do you want to skip `checkAndTransitionHardForkByNumber` if the opts.consensus if present in constructor? (might need a flag) or one can also provide this fn as a constructor arg.

Reply:
Actually, nm, it already exists.  [hardforkByHeadBlockNumber](https://github.com/ethereumjs/ethereumjs-monorepo/blob/f4cb2524210d1b7af55d4cd2c0004aee1966b01f/packages/blockchain/src/types.ts#L62)

This is only set to true in the `client` as far as I can tell. If we want the blockchain to transition hardforks for custom consensus types, we should probably just remove the `default` behaviour in the switch statement to throw if an unsupported consensus algorithm is provided.  Thoughts?

Reply:
yes we can ignore default behavior if custom consensus is provided :+1: